src/site/topic2-cpp.rst

```diff
@@ -0,0 +1,648 @@
+    int someFunction(int a, int b){
+        int c = a + b;
+        return c;
+    }
+
+    int main(){
+        std::cout << someFunction(1, 2) << std::endl;
+    }
+
+* You will see that the Java and C++ examples are very similar
+* The one catch here is that the function is defined before main
+    * We can put it after main if really we want, but then we would need to add a function prototype before the main
+    * ``int someFunction(int, int);	// Function Prototype``
+
+
+Temperature Converter
+---------------------
+
+**Python**
+
+.. code-block:: Python
+    :linenos:
+    :emphasize-lines: 3
+
+    # Python --- Convert farenheit to celcius
+    def fahrenheit_to_celcius (fahrenheit):
+        celsius = (fahrenheit - 32) * 5/9
+        return celsius
+
+
+**Java**
+
+.. code-block:: Java
+    :linenos:
+    :emphasize-lines: 3
+
+    // Java --- Convert farenheit to celcius
+    static double fahrenheitToCelcius(double fahrenheit) {
+        double celsius = (fahrenheit - 32) * 5.0/9.0;
+        return celsius;
+    }
+
+
+**C++**
+
+.. code-block:: cpp
+    :linenos:
+    :emphasize-lines: 2, 8
+
+    // C++ --- Convert farenheit to celcius
+    double FtoC(double fahrenheit){
+        double celsius = (fahrenheit - 32) * 5.0/9.0;
+        return celsius;
+    }
+
+
+Comments
+========
+
+.. code-block:: cpp
+    :linenos:
+
+    // This is a single line comment in C++
+
+    /*
+    This is a
+    multi line
+    comment in
+    C++
+     */
+
+    /**
+     * Convert the provided temperature from fahrenheit
+     * to celsius.
+     *
+     * This also demonstrates how to write a javadoc
+     * comment.
+     *
+     * @param fahrenheit    temperature in fahrenheit
+     * @return              temperature in celsius
+     */
+    double FtoC(double fahrenheit){
+        double celsius = (fahrenheit - 32) * 5.0/9.0;
+        return celsius;
+    }
+
+
+Booleans
+========
+
+* Python --- ``someBool = True``
+* Java --- ``boolean someBool = true;``
+* C++ --- ``bool someBool = true;``
+
+.. warning::
+
+    Although C++ has booleans, you will see the value ``1`` mean ``true``, and ``0`` mean ``false`` (or, really, anything
+    *not* ``1`` is ``false``).
+
+
+Conditionals
+------------
+
+**Python**
+
+.. code-block:: Python
+    :linenos:
+
+    # Python --- if/else
+    # Assume someBool exists and is a boolean
+    if someBool:
+        print("it was true")
+    else:
+        print("it was false")
+
+
+**Java**
+
+.. code-block:: Java
+    :linenos:
+
+    // Java --- if/else
+    // Assume someBool exists and is a boolean
+    if (someBool) {
+        System.out.println("it was true");
+    } else {
+        System.out.println("it was false");
+    }
+
+
+**C++**
+
+.. code-block:: cpp
+    :linenos:
+
+    // C++ --- if/else
+    // Assume someBool exists and is a boolean
+    if (someBool) {
+        std::cout << "it was true" << std::endl;
+    } else {
+	    std::cout << "it was false" << std::endl;
+    }
```

Comment:
Minor, spacing is off here.

Reply:
It's weird. It seems correct in my local editor. I'll try anyways./

Reply:
Maybe tab characters snuck in?